### ISSUE TITLE
[Request] External DB editing clarification

### ISSUE THREAD

**Issue by RossAscends** [kalilistic/PlayerTrack#174]
I'm trying to use SQL queries to automate some changes in player tracker's db

As a simple example. 

```sql
INSERT INTO player_categories (created, updated, player_id, category_id)
SELECT p.created, p.updated, p.id, 1 --get the necessary data from players to fill out the new row in player_categories
FROM players p
LEFT JOIN player_categories pc ON p.id = pc.player_id --skip pre-existing matches
WHERE pc.player_id IS NULL --only act when there is no match
AND SUBSTR(p.customize, 2, 1) = X'00'; --insert if 2nd customize byte is 0
```

This *should* create new entries in `player_categories` for any row in `players` which:
-  is not already present in `player_categories` (by checking for preexisting `player_id`) 
- the `customize` column has 00 as the second byte (indicating a male character)

If those conditions are met, the new row is made and the `category_id` is set to 1

Upon running this Query, I can see the size of `player_categories` increase, but the changes don't seem to be reflected in game immediately. I've tried logging off and back in, thinking this might cause PT to re-read the database, but No-go. 

Any idea?  Are there some other checks that PT does to verify DB entries before implementing the changes in-game?

**Comment by RossAscends** [comment]
Ok it works if the game is totally closed out and restarted (not just log off). However, now ptracker takes 7 minutes to load. Whew.

**Comment by kalilistic** [comment]
yeah, not safe to edit the DB while it's running (records will be overwritten) due to heavy caching in the plugin. I wonder if it was creating additional records that first run (e.g. new config records).

**Comment by RossAscends** [comment]
I further expanded my Query effects to cover all 25,364 players in my db (at the moment), assigning each of them to a category if they were not already.

After doing that it took 28 minutes for the PT UI to display when FF was next booted up after the Query. Subsequent loads are instantaneous still tho, so that's good!

All the while it was writing to my SSD C drive where XIVLauncher is installed. 
I assume it was doing something in a temp file in /appData/ somewhere, but couldn't figure out where. 

There's no abnormally large files in the /pluginConfig/ folder either. The new db is only 1mb larger than it was before I did any Query work at all, but does have about 19k more categories entries to handle. 

FFXIV itself is installed on a NVME on D:\ , which should be faster..maybe? 

Is there any way to change where XIV handles its appData?

**Comment by kalilistic** [comment]
I believe there's a program argument but I'm not too sure - would need to check the xiv launcher discord.
